packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx

```diff
@@ -14,42 +14,64 @@ import { observer } from '@deriv/stores';
 import { Localize, localize } from 'Components/i18next';
 import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
 import MyProfileSeparatorContainer from 'Components/my-profile/my-profile-separator-container';
+import { buy_sell } from 'Constants/buy-sell';
+import { ad_type } from 'Constants/floating-rate';
 import { TAdvert } from 'Types';
-import ShareMyAdsIcons from './share-my-ads-socials';
 import ShareMyAdsCard from './share-my-ads-card';
+import ShareMyAdsSocials from './share-my-ads-socials';
 
 const ShareMyAdsModal = ({ advert }: TAdvert) => {
     const [is_copied, copyToClipboard, setIsCopied] = useCopyToClipboard();
+    const { account_currency, advertiser_details, id, local_currency, rate_display, rate_type, type } = advert;
+    const { id: advertiser_id } = advertiser_details;
+    const { hideModal, is_modal_open } = useModalManagerContext();
 
     const divRef = React.useRef(null);
-    // TODO: replace with proper url when available
-    const advert_url = window.location.href;
-
-    const { hideModal, is_modal_open } = useModalManagerContext();
+    const advert_url = `${window.location.origin}/cashier/p2p/advertiser?id=${advertiser_id}&advert_id=${id}`;
```

Comment:
There is a function called `websiteUrl` from shared to get the current host, maybe you can use it. You can ignore this comment 😄 

```suggestion
    const advert_url = `${websiteUrl()}cashier/p2p/advertiser?id=${advertiser_id}&advert_id=${id}`;
```